md-to-toc.py

```diff
@@ -28,7 +28,7 @@ def to_github_anchor(title):
     anchor_name = title.strip().lower().replace(' ', '-')
 
     # Strip all invalid characters
-    anchor_name = re.sub("[^A-Za-z0-9\-_]", "", anchor_name)
+    anchor_name = re.sub("[\]\[\!\"\#\$\%\&\'\(\)\*\+\,\.\/\:\;\<\=\>\?\@\\\^\_\{\|\}\~]", "", anchor_name)
```

Comment:
This is a useful change, but could we make this code more clear by using Python raw strings here? Could you try with:

```python
    anchor_name = re.sub(r"[][!"#$%&'()*+,./:;<=>?@\^_{|}~]", "", anchor_name)
```

I haven't tested this, but I assume it should give the same result.